File: js/task3/S3_PIPE_01.js
function pipe(...fns) {
  return function (value) {
    let res = value;

    for (let i = 0; i < fns.length; i++) {
      res = fns[i](res);
    }

    return res;
  };
}

// Tests
const add1 = (x) => x + 1;
const double = (x) => x * 2;

const run = pipe(add1, double);
console.log(run(3)); // 8

File: python/stage3/S3_PIPE_01.py
def pipe(*fns):
    def apply(x):
        for fn in fns:
            x = fn(x)
        return x
    return apply


# Tests
f = pipe(lambda x: x + 1, lambda x: x * 2)
print(f(3))  # (3 + 1) * 2 = 8
